### PR TITLE
fpc: Add ppc64le support.

### PR DESCRIPTION
**Pull request by backerman**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [X] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [X] I built this PR locally for my native architecture, (ppc64le-libc)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

Details: Ran test suite. There are a small number of failures (36 of 7k+). The tests that fail are the same ones that the bootstrap compiler fails, so I'm fairly sure that no additional failures are introduced by this packaging template.

### CODE REVIEW

File: srcpkgs/fpc/template [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 version=3.2.0
 revision=1
-archs="x86_64* i686*"
+archs="x86_64* i686* ppc64le"
 create_wrksrc=yes
 build_wrksrc="${pkgname}build-${version}"
 conf_files="/etc/fpc.cfg /etc/fppkg.cfg"
 hostmakedepends="rpmextract"
-makedepends="ncurses-devel zlib-devel expat-devel"
+makedepends="ncurses-devel zlib-devel expat-devel bsdtar"
```
Comment (ericonr) [inline srcpkgs/fpc/template:10]: This is available by default in the build system.

File: common/shlibs [hunk 2]
```diff
@@ -3974,3 +3974,4 @@ libmd.so.0 libmd-1.0.3_1
 libldacBT_abr.so.2 ldacBT-2.0.2.3_1
 libldacBT_enc.so.2 ldacBT-2.0.2.3_1
 libgumbo.so.1 gumbo-parser-0.10.1_2
+libpas2jlib.so fpc-3.2.0_1
```
Comment (ericonr) [inline common/shlibs:3977]: If you're just solving the build time warning, this isn't necessary.

**Review by ericonr — COMMENTED** [review]
@q66

File: srcpkgs/fpc/template [hunk 3]
```diff
@@ -2,12 +2,12 @@
 pkgname=fpc
 version=3.2.0
 revision=1
-archs="x86_64* i686*"
+archs="x86_64* i686* ppc64le"
```
[the remark this answers is not in the document]
Comment (q66) [reply srcpkgs/fpc/template:5]: apparently not, trying on musl, getting errors:

```
/usr/bin/ld: /builddir/fpc-3.2.0/fpcbuild-3.2.0/fpcsrc/rtl/units/powerpc64-linux/si_c.o:(.data.n_TC_$SI_C_$$_START_ADDRESSES+0x10): undefined reference to `__libc_csu_init'
/usr/bin/ld: /builddir/fpc-3.2.0/fpcbuild-3.2.0/fpcsrc/rtl/units/powerpc64-linux/si_c.o:(.data.n_TC_$SI_C_$$_START_ADDRESSES+0x18): undefined reference to `__libc_csu_fini'
```

[the remark this answers is not in the document]
Comment (q66) [reply srcpkgs/fpc/template:5]: looks like we patch that out on x86_64-musl, i wonder if the same could be done here

[the remark this answers is not in the document]
Comment (q66) [reply srcpkgs/fpc/template:5]: i think the patch we have is actually broken and probably doesn't really work at all... so keep it restricted to non-musl is the right thing for now

**Pull request closed without merging**